dvc/remote/base.py

```diff
@@ -966,12 +978,14 @@ def checkout(
                         self.path_info, checksum, filter_info
                     ),
                 )
-            return failed
+            if failed:
+                raise CheckoutError([failed])
```

Comment:
A lot of tests will fail because of this and I'm really not sure if this is a good thing or bad. But, for now, it really makes the control flow easier.

Reply:
Tests seem to have passed. :see_no_evil: